### PR TITLE
Add Rx contract-adhering error handling.

### PR DESCRIPTION
**Pull request by JakeWharton**

### CODE REVIEW

File: src/test/java/com/jakewharton/retrofit2/adapter/rxjava2/FlowableThrowingTest.java [hunk 1]
```diff
@@ -0,0 +1,293 @@
+/*
+ * Copyright (C) 2015 Square, Inc.
```
Comment (vanniktech) [inline src/test/java/com/jakewharton/retrofit2/adapter/rxjava2/FlowableThrowingTest.java:2]: nit: should not this be also 2016?

File: src/test/java/com/jakewharton/retrofit2/adapter/rxjava2/ObservableThrowingTest.java [hunk 2]
```diff
@@ -0,0 +1,288 @@
+/*
+ * Copyright (C) 2015 Square, Inc.
```
Comment (vanniktech) [inline src/test/java/com/jakewharton/retrofit2/adapter/rxjava2/ObservableThrowingTest.java:2]: same here

File: src/test/java/com/jakewharton/retrofit2/adapter/rxjava2/SingleThrowingTest.java [hunk 3]
```diff
@@ -0,0 +1,240 @@
+/*
+ * Copyright (C) 2015 Square, Inc.
```
Comment (vanniktech) [inline src/test/java/com/jakewharton/retrofit2/adapter/rxjava2/SingleThrowingTest.java:2]: same here

File: src/test/java/com/jakewharton/retrofit2/adapter/rxjava2/FlowableThrowingTest.java (hunk 1 above)
Comment (JakeWharton) [inline src/test/java/com/jakewharton/retrofit2/adapter/rxjava2/FlowableThrowingTest.java:2]: The completable test was written from scratch. These ones were copied from existing source files and adapted which is why they retain their original copyright date.

**Pull request merged**